### DISCUSSION TITLE
Gunicorn Command Args

### DISCUSSION THREAD

**Discussion by LeeJCherry** [apache/superset#18175] [Q&A / Help]
Looking for some help please

We have embedded Superset into our UI's for reports and getting intermittant HTTP 502 errors loading some widgets.  If I refresh the page it loads and also, it's random, a page with 5 widgets for example, widget 4 will throw a 502, if I refresh the page, all 5 will load fine, if I refresh again, widget 1 will fail to load and the other 4 are fine, it's a all over the place.

Superset is hosted in Kubernetes and I have traced the request though Kubernetes ingress which is reporting that it's Superset throwing the 502 error.  I have searched the Superset logs and I am not seeing any 502 error at all and the next step of my investigation is to dig into the Gunicorn logs, to see if there is anything being reported there however, I can't see any logs for Gunicorn.  Running `ps -aux` gives me this

````
USER       PID %CPU %MEM    VSZ   RSS TTY      STAT START   TIME COMMAND
root         1  0.0  0.0   2376   628 ?        Ss   Jan24   0:00 /bin/sh -c . /app/pythonpath/superset_bootstrap.sh; /usr/bin/docker-entrypoint.sh
root       141  0.0  0.0   5476  1612 ?        S    Jan24   0:00 /bin/bash /usr/bin/docker-entrypoint.sh
root       142  0.0  0.2  30636 19636 ?        S    Jan24   0:19 /usr/local/bin/python /usr/bin/gunicorn --bind 0.0.0.0:8088 --access-logfile - --error-logfile - --workers 1 --worker-class gthread --threads 20 --timeout 60 --limit-request-line 0 --limit-request-field_size
root       145  0.1  4.2 1768440 340580 ?      Sl   Jan24   5:04 /usr/local/bin/python /usr/bin/gunicorn --bind 0.0.0.0:8088 --access-logfile - --error-logfile - --workers 1 --worker-class gthread --threads 20 --timeout 60 --limit-request-line 0 --limit-request-field_size
root       173  0.0  0.0   5740  2268 pts/0    Ss   12:05   0:00 bash
root       193  0.0  0.0   9380  1460 pts/0    R+   12:42   0:00 ps -aux
```
 Judging by this part I don't think logging is even enabled?

--access-logfile - --error-logfile - 

This then took me to discover this via Google `GUNICORN_CMD_ARGS`.  Do I need to set this to enable logging and if so, is that even possible without rebuilding my own Docker image?

**Comment by andrey-pervushin** [comment]
up!
